watertap/tools/parameter_sweep.py

```diff
@@ -319,108 +338,239 @@ def _interp_nan_values(global_values, global_results):
+    # Store the inputs
+    for key in sweep_params.keys():
+        var = sweep_params[key].pyomo_object
+        var_str = sweep_params[key].pyomo_object.name
+        var_str_list.append(var_str)
+        output_dict["sweep_params"][var_str] =  _create_component_output_skeleton(var, num_samples)
 
-        sweep_params: A dictionary containing the values to vary with the format
-                      ``sweep_params['Short/Pretty-print Name'] =
-                      (model.fs.variable_or_param[index], lower_limit, upper_limit, num_samples)``.
-                      A uniform number of samples ``num_samples`` will be take between
-                      the ``lower_limit`` and ``upper_limit``.
+    if outputs is None:
+        # No outputs are specified, so every Var, Expression, and Objective on the model should be saved
+        for pyo_obj in model.component_data_objects((pyo.Var, pyo.Expression, pyo.Objective), active=True):
+            # Only need to save this variable if it isn't one of the value in sweep_params
+            if pyo_obj.name not in var_str_list:
+                output_dict["outputs"][pyo_obj.name] = _create_component_output_skeleton(pyo_obj, num_samples)
+    else:
+        # Save only the outputs specified in the outputs dictionary
+        for pyo_obj in outputs.values():
+            output_dict["outputs"][pyo_obj.name] = _create_component_output_skeleton(pyo_obj, num_samples)
 
-        outputs : A dictionary containing "short names" as keys and and Pyomo objects
-                  on ``model`` whose values to report as values. E.g.,
-                  ``outputs['Short/Pretty-print Name'] = model.fs.variable_or_expression_to_report``.
+    return output_dict
 
-        results_file (optional) : The path and file name where the results are to be saved;
-                                   subdirectories will be created as needed.
+# ================================================================
 
-        optimize_function (optional) : A user-defined function to perform the optimization of flowsheet
-                                       ``model`` and loads the results back into ``model``. The first
-                                       argument of this function is ``model``\. The default uses the
-                                       default IDAES solver, raising an exception if the termination
-                                       condition is not optimal.
+def _create_component_output_skeleton(component, num_samples):
+
+    comp_dict = {}
+    comp_dict["value"] = np.zeros(num_samples, dtype=np.float)
+    if hasattr(component, 'lb'):
+        comp_dict["lower bound"] = component.lb
+    if hasattr(component, 'ub'):
+        comp_dict["upper bound"] = component.lb
+    if hasattr(component, 'get_units'):
+        unit_obj = component.get_units()
+        if unit_obj is not None:
+            comp_dict["units"] = component.get_units().name
+        else:
+            comp_dict["units"] = "non-dimensional"
```

Comment:
We might just say "None". All we know in this case is that the user didn't *specify* units, not that the component doesn't have a unit in actually (i.e., is *dimensionless*, like ratio).

Reply:
Changed non-dimensional to "None" in commit 2e87f5633e2ccb6bc512089d5cbb6703cb3a16fd